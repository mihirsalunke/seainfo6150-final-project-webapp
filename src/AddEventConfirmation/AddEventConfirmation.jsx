import React from "react";
import PropTypes from "prop-types";
import styles from "./AddEventConfirmation.module.css";

const AddEventConfirmation = (props) => {
    return (
        <div className={styles.container}>
            <div className={styles.confirmationMessage}>
                <p>Thank You..!! <br/>Event <b>{props.eventName}</b> successfully created</p>
            </div>

            <div className={styles.eventDetails}>
                <div className={styles.title}>
                    <p>Event Details:</p>
                </div>

                <div className={styles.text}>
                    <p>Event Name: <span>{props.eventName}</span></p>
                </div>
                <div className={styles.text}>
                    <p>Event Date: <span>{props.eventDate}</span></p>
                    <p>Event Time: <span>{props.eventTime}</span></p>
                </div>
                <div className={styles.location}>
                    <p>Event Location: <span>{props.eventLocation}</span></p>
                </div>
            </div>
        </div>
    );
};

AddEventConfirmation.propTypes = {
    eventName: PropTypes.string.isRequired,
    eventDate: PropTypes.string.isRequired,
    eventTime: PropTypes.string.isRequired,
    eventLocation: PropTypes.string.isRequired,
};

export default AddEventConfirmation;