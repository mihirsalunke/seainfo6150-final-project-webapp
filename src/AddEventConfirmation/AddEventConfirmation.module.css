@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background: white;
    min-width: 700px;
    border: 1px solid black;
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-radius: 10px;
    position: fixed;
    top: 40%;
    left: 50%;
    transform: translate(-50%, -40%);
}

.confirmationMessage {
    font-size: 2em;
    margin: 2%;
    color: rgb(33, 141, 6);
    text-align: center;
}

.confimationMessage b {
    font-family: Rowdies, sans-serif;
}

.eventDetails {
    border: 1px solid black;
    border-radius: 7px;
    margin: 2%;
}

.title {
    text-align: center;
    font-family: Rowdies, sans-serif;
}

.text {
    padding-left: 5%;
}

.text span {
    font-family: "Nerko One", sans-serif;
}

.location {
    padding-left: 5%;
}

.location span {
    font-family: "Nerko One", sans-serif;
}