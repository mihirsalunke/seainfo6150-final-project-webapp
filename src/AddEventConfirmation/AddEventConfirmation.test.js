import React from "react";
import AddEventConfirmation from "./AddEventConfirmation.jsx";
import { render } from "@testing-library/react";

const eventName = "Future of AI";
const eventDate = "2020-12-24";
const eventTime = "1:00 pm";
const eventLocation = "Chicago";

describe("AddEventConfirmation tests", () => {
    it("renders correctly", () => {
      const { container } = render(<AddEventConfirmation eventName={eventName} eventDate={eventDate} eventTime={eventTime} eventLocation={eventLocation} />);
      expect(container).toMatchSnapshot();
    });
});