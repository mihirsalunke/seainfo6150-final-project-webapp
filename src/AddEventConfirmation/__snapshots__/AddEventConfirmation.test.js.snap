// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEventConfirmation tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="confirmationMessage"
    >
      <p>
        Thank You..!! 
        <br />
        Event 
        <b>
          Future of AI
        </b>
         successfully created
      </p>
    </div>
    <div
      class="eventDetails"
    >
      <div
        class="title"
      >
        <p>
          Event Details:
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Event Name: 
          <span>
            Future of AI
          </span>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Event Date: 
          <span>
            2020-12-24
          </span>
        </p>
        <p>
          Event Time: 
          <span>
            1:00 pm
          </span>
        </p>
      </div>
      <div
        class="location"
      >
        <p>
          Event Location: 
          <span>
            Chicago
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
